Ongoing PR

jackviel opened the pull request:
Worked on implementing search with the books, movies, and music APIs, worked on the Results Adapter, Results Model, API connectivity, and results rendering when searching in the review compose fragment with the end goal of searching all 3 APIs on the same SearchView to show books, movies, and music based on the same query.

When a query is entered in the SearchView, it'll search through each API, then it'll add the results to a results array, and then append the results to the results adapter RecyclerView.

- Tested for queries that should bring up results for only one API, to check if there aren't any results in one API, it should only add the correct results.
- Also tested for queries that gather less than the minimum amount of results displayed in the RecyclerView results.

jackviel committed:
Search implemented for books API

jackviel committed:
added the movies api client and implemented search for movies in the same search view as the books

jackviel committed:
music api search implemented

jackviel committed:
attached media to review post

jackviel committed:
made selected media details show up on the feed, also added a review title field in the compose fragment.

jackviel committed:
made selected media show up on the feed, also added a review title field in the compose fragment

brianwal commented on app/src/main/java/com/example/captstone/ResultAdapter.java:
```diff
@@ -0,0 +1,113 @@
+    }
+
+    // Define the method that allows the parent activity or fragment to define the listener
+    public void setOnItemClickListener(OnItemClickListener listener) {
```
Is it possible to set this in the constructor?

brianwal replied:
If not do you need to handle error case when already set

brianwal commented on app/src/main/java/com/example/captstone/ResultAdapter.java:
```diff
@@ -0,0 +1,113 @@
+        View resultView = inflater.inflate(R.layout.item_result, parent, false);
+
+        // Return a new holder instance
+        ResultAdapter.ViewHolder viewHolder = new ResultAdapter.ViewHolder(resultView, listener);
```
Do you need to handle listener == null case?

brianwal commented on app/src/main/java/com/example/captstone/fragments/ComposeFragment.java:
```diff
@@ -5,29 +5,52 @@
     private EditText etBody;
     private Button bSubmit;
     public ProgressBar pb;
+    public SearchView svMedia;
```
No using consistant style in code

brianwal commented on app/src/main/java/com/example/captstone/fragments/ComposeFragment.java:
```diff
@@ -43,6 +66,74 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
+            public void onItemClick(View itemView, int position) {
+                Toast.makeText(
+                        getContext(),
+                        "An item at position " + position + " clicked!",
```
String format over multiple concat

brianwal commented on app/src/main/java/com/example/captstone/fragments/ComposeFragment.java:
```diff
@@ -43,6 +66,74 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
+                            }
+                        } catch (JSONException e) {
+                            // Invalid JSON format, show appropriate error.
+                            e.printStackTrace();
```
poor error handling

brianwal commented on app/src/main/java/com/example/captstone/models/Result.java:
```diff
@@ -0,0 +1,95 @@
+import java.util.ArrayList;
+
+public class Result {
+    private String openLibraryId;
```
final

brianwal commented on app/src/main/java/com/example/captstone/models/Result.java:
```diff
@@ -0,0 +1,95 @@
+            JSONObject resultJson;
+            try {
+                resultJson = jsonArray.getJSONObject(i);
+            } catch (Exception e) {
```
Not good practice to catch Exception.  Catch specific exceptions

jackviel committed:
added functionality to add media from reviews in the feed to a netflix-like "my list"

jackviel committed:
implemented double tap gesture to button to add media to user's "my list"

jackviel committed:
Implemented feed cache when offline & implemented PR feedback

jackviel committed:
implemented feed cache for offline use

jackviel committed:
started implementing material design, worked on overall styling, fixed feed query lag & keyboard focus on compose fragment, cleaned up imports on various files, added tab layout on the home fragment & profile fragment.

jackviel committed:
Started implementing the trending fragment replacing the compose fragment.

Currently just shows the currently trending movies, songs, and books.

Had to create new client classes to get different data from the APIs, and had to incorporate a new API (NY Times API) to get the currently trending books (NY Best Seller's).

Currently the trending fragment is a recyclerview with the fetched trending media.

jackviel committed:
added a tab layout for the trending fragment with a viewpager under populated with the former trending feed fragment, worked on styling, fixed trending book title capitalization, worked on overall styling

jackviel committed:
implemented trending media item page

jackviel committed:
implemented 'my reviews' tab fragment in profile fragment, organized files

jackviel committed:
implemented actionbar media search

jackviel committed:
implemented functionality to submit reviews on the SelectedMediaActivity

jackviel committed:
fixed creator on trending songs, cleaned up comments

jackviel committed:
made log TAGs and database in ParseApplication private

jackviel committed:
implemented rating bar functionality

jackviel committed:
worked on displaying the 'my-list' on user profile tab and implemented the 'add to my-list' button & indicator on main review feed

jackviel committed:
implemented change username function
